### PR TITLE
Fix CupertinoTextSelectionToolbar clipping

### PR DESCRIPTION
**Pull request by luccasclezar**
The CupertinoTextSelectionToolbar sets the maxWidth of the whole toolbar to the width of the first page. This ends up clipping other pages from the toolbar. This PR just removes this limitation.

It was easy enough that I thought there was a catch, but I ran the tests locally and they all passed.

|Before|After|
|-|-|
|Simulator Screenshot - iPhone Xʀ - 2023-11-09 at 19 45 29|Simulator Screenshot - iPhone Xʀ - 2023-11-09 at 19 44 30|

https://github.com/flutter/flutter/issues/138177

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by luccasclezar** [comment]
@LongCatIsLooong Whew, glad this was really this easy to fix 😅 Will add a test as soon as possible.

**Comment by LongCatIsLooong** [comment]
Hello from the triage meeting, are you planning to continue working on this PR?

**Comment by luccasclezar** [comment]
@LongCatIsLooong Sorry, I couldn't find enough time yet to implement the tests, I will try to do it today.

**Comment by luccasclezar** [comment]
@LongCatIsLooong Done! I reverted my previous commit locally to make sure the test failed without the fix, so everything should be in order.

**Review by LongCatIsLooong — APPROVED** [review]
The change LGTM. Thank you for contributing! The ci fails are not related, I've synced the PR branch via the github UI hopefully that will do it.

File: packages/flutter/test/cupertino/text_selection_toolbar_test.dart [hunk 1]
```diff
@@ -274,6 +274,52 @@ void main() {
...
+                SizedBox(width: 100),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    Finder findSizedBoxWithWidth(double width) =>
+      find.byWidgetPredicate((Widget widget) => widget is SizedBox && widget.width == width);
```
Comment (justinmc) [inline packages/flutter/test/cupertino/text_selection_toolbar_test.dart:299]: I think you should remove the first `=>` based on https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods.

File: packages/flutter/test/cupertino/text_selection_toolbar_test.dart [hunk 2]
```diff
@@ -274,6 +274,52 @@ void main() {
...
+                SizedBox(width: 100),
+                SizedBox(width: 100),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    Finder findSizedBoxWithWidth(double width) =>
```
Comment (justinmc) [inline packages/flutter/test/cupertino/text_selection_toolbar_test.dart:298]: Nit: It's probably better to declare a GlobalKey for each of the SizedBoxes and find them by key.

File: packages/flutter/test/cupertino/text_selection_toolbar_test.dart [hunk 3]
```diff
@@ -274,6 +274,52 @@ void main() {
...
+    expect(findSizedBoxWithWidth(300), findsNothing);
+
+    // Show the next page.
+    await tester.tapAt(tester.getCenter(findOverflowNextButton()));
+    await tester.pumpAndSettle();
+
+    // The second page should show only the second button, but it should not
+    // limit its width.
+    // expect(findSizedBoxWithWidth(100), findsNothing);
+    // expect(findSizedBoxWithWidth(300), findsOneWidget);
```
Comment (justinmc) [inline packages/flutter/test/cupertino/text_selection_toolbar_test.dart:312]: Why are these commented out?

File: packages/flutter/test/cupertino/text_selection_toolbar_test.dart [hunk 4]
```diff
@@ -274,6 +274,52 @@ void main() {
...
+    expect(tester.getSize(findSizedBoxWithWidth(300)).width, 300);
+
+    // Show the next page
```
Comment (justinmc) [inline packages/flutter/test/cupertino/text_selection_toolbar_test.dart:315]: Missing a period here.

**Review by justinmc — COMMENTED** [review]
When I wrote this I deliberately kept it to the first page's width because that's what I saw native iOS doing. It looks like that's no longer the case though, so thank you for changing it!

**Comment by justinmc** [comment]
@luccasclezar Also FYI you have a merge conflict.

**Comment by justinmc** [comment]
@luccasclezar Are you still planning to come back to this PR?

**Comment by luccasclezar** [comment]
@justinmc Yep, I'm just very busy currently 😅 I will try to improve what you suggested tomorrow, and if I get any complications I will finish on the weekend.

**Comment by luccasclezar** [comment]
@justinmc Okay, I did everything you suggested and resolved the conflict. Everything should be working now 😁

**Review by justinmc — APPROVED** [review]
LGTM 👍 . Thanks for the fixes!

**Comment by justinmc** [comment]
@luccasclezar Can you update the branch with master? I think that will fix the broken "ci.yaml validation" check...

**Comment by LongCatIsLooong** [comment]
```
   info • Missing type annotation • packages/flutter/test/cupertino/text_selection_toolbar_test.dart:277:5 • always_specify_types
   info • Missing type annotation • packages/flutter/test/cupertino/text_selection_toolbar_test.dart:278:5 • always_specify_types
   info • Missing type annotation • packages/flutter/test/cupertino/text_selection_toolbar_test.dart:279:5 • always_specify_types
   info • Missing type annotation • packages/flutter/test/cupertino/text_selection_toolbar_test.dart:280:5 • always_specify_types
```

**Comment by luccasclezar** [comment]
@LongCatIsLooong I can fix these tomorrow

**Comment by luccasclezar** [comment]
@LongCatIsLooong Done 👍

**Comment by justinmc** [comment]
There are now some analyzer errors here.

**Pull request merged**